ister_gui.py

```diff
@@ -641,24 +643,215 @@ def validate_condition(self):
+        static_button = urwid.AttrMap(static_button, None,
+                                      focus_map='reversed')
+        reset_button = urwid.AttrMap(reset_button, None,
+                                     focus_map='reversed')
+        proxy_button = urwid.AttrMap(proxy_button, None,
+                                     focus_map='reversed')
+        self.static_col = urwid.Columns([static_button, urwid.Divider()])
+        self.reset_col = urwid.Columns([reset_button, urwid.Divider()])
+        self.proxy_col = urwid.Columns([proxy_button, urwid.Divider()])
+        self.installer_proxy = urwid.Edit('installer proxy (https): ', '')
+        self.static_ip = None
+        self.interface = None
+        self.gateway = None
+        self.error = None
+        self.config = None
+        self.ifaceaddrs = None
+
     def handler(self, config):
-        alert_content = 'Step 1 of 5\n\n'                                     \
-                        'Network Requirements:\n'                             \
-                        '  * Wired network connection\n'                      \
-                        '  * DHCP\n'                                          \
-                        '  * Internet Access\n\n'                             \
-                        'The installer will navigate most environments with ' \
-                        'autoproxies automatically\n'
-
-        alert = Alert("Clear Linux OS for Intel Architecture Installer",
-                      alert_content)
-        alert.do_alert()
-        return alert.response
+        # make config an insance variable so we can copy proxy settings to it
+        self.config = config
+
+        # normally much of this would belong in __init__, but we want it
+        # refreshed every time the user refreshes/returns to the screen
+        self.build_ui_widgets()
+        self.build_ui()
+        self._action = self.run_ui()
+        if self._action == 'Refresh':
+            # re-build the widgets
+            return self._action
+
+        self.error = ''
+        try:
+            ipaddress.ip_address(self.static_ip.get_edit_text())
+        except Exception as err:
+            self.error = 'The configuration for "ip" is invalid {}'.format(err)
+            # an empty action signals to refresh the page
+            self._action = ''
+        if self.interface.get_edit_text() not in self.ifaceaddrs.keys():
+            self.error = 'Interface "{}" not detected'\
+                         .format(self.interface.get_edit_text())
+            # an empty action signals to refresh the page
+            self._action = ''
+
+        if not self.error:
+            config = self.config
+            return self._action
+        else:
+            Alert('Error!', self.error).do_alert()
+            self.error = ''
+            return self._action
+
+    def build_ui_widgets(self):
+        """Build ui handler
+        The last urwid Divider helps the tab movement to work"""
```

Comment:
I thought it was fine in methods. PEP8 doesn't complain about them. I can add a blank line if you want me too.